### ISSUE TITLE
[BUG] 0.41.0 segfault in ZTS build

### ISSUE THREAD

**Issue by remicollet** [DataDog/dd-trace-php#782]
```
(gdb) run -n -d extension=/dev/shm/BUILD/php-pecl-datadog-trace-0.41.0/ZTS/modules/ddtrace.so -m
rogram received signal SIGSEGV, Segmentation fault.
0x00007ffff75db262 in zm_startup_ddtrace (type=<optimized out>, module_number=15) at /dev/shm/BUILD/php-pecl-datadog-trace-0.41.0/ZTS/src/ext/ddtrace.c:137
137	    ddtrace_bgs_log_minit(PG(error_log));

(gdb) bt
#0  0x00007ffff75db262 in zm_startup_ddtrace (type=<optimized out>, module_number=15) at /dev/shm/BUILD/php-pecl-datadog-trace-0.41.0/ZTS/src/ext/ddtrace.c:137
#1  zm_startup_ddtrace (type=<optimized out>, module_number=15) at /dev/shm/BUILD/php-pecl-datadog-trace-0.41.0/ZTS/src/ext/ddtrace.c:123
#2  0x0000555555818260 in zend_startup_module_ex (module=0x555555a83710) at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/Zend/zend_API.c:1878
#3  zend_startup_module_ex (module=0x555555a83710) at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/Zend/zend_API.c:1831
#4  0x0000555555818310 in zend_startup_module_zval (zv=<optimized out>) at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/Zend/zend_API.c:1893
#5  0x00005555558252cb in zend_hash_apply (ht=ht@entry=0x555555a1ed80 <module_registry>, apply_func=apply_func@entry=0x555555818300 <zend_startup_module_zval>)
    at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/Zend/zend_hash.c:1689
#6  0x000055555581861e in zend_startup_modules () at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/Zend/zend_API.c:2004
#7  0x00005555557a7636 in php_module_startup (sf=<optimized out>, additional_modules=<optimized out>, num_additional_modules=<optimized out>)
    at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/main/main.c:2342
#8  0x00005555558a89d1 in php_cli_startup (sapi_module=<optimized out>) at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/sapi/cli/php_cli.c:420
#9  0x0000555555666403 in main (argc=5, argv=0x555555a22210) at /usr/src/debug/php-7.3.15-1.fc31.remi.x86_64/sapi/cli/php_cli.c:1360

```

**Comment by remicollet** [comment]
You cannot play with globals in MINIT (per-process), but only in GINIT (per-thread)

**Comment by morrisonlevi** [comment]
Thanks for the report, Remi.

**Comment by morrisonlevi** [comment]
Remi, this should be fixed in 0.41.1; feel free to re-open if this is not the case.
